Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererBase.cs

```diff
@@ -355,8 +363,80 @@ private void RenderDataCell (BocColumnRenderingContext<TBocColumnDefinition> ren
 
       renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Td);
 
+      var validationInfoText = renderingContext.Control.GetResourceManager().GetString(BocList.ResourceIdentifier.ValidationErrorInfoCell);
+      var validationFailures = renderingContext.Control.ValidationFailureRepository.GetUnhandledValidationFailuresForDataCell(
+          arguments.BusinessObject,
+          renderingContext.ColumnDefinition,
+          false);
+
+      if (validationFailures.Count > 0)
+      {
+        var tooltipStringBuilder = new StringBuilder();
+        tooltipStringBuilder.AppendLine(validationInfoText);
+        foreach (var validationFailure in validationFailures)
+        {
+          tooltipStringBuilder.Append(" • ");
+          tooltipStringBuilder.AppendLine(validationFailure.Failure.ErrorMessage);
+        }
+
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Id, BocRowRenderer.GetCellIDForValidationMarker(renderingContext.Control, arguments.RowIndex, renderingContext.VisibleColumnIndex));
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Title, tooltipStringBuilder.ToString());
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, CssClasses.ValidationErrorMarker);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        var iconInfo = new IconInfo(_resourceUrlFactory.CreateThemedResourceUrl(typeof(InfrastructureResourceUrlFactory), ResourceType.Image, c_validationIndicatorIcon).GetUrl());
+        iconInfo.Render(renderingContext.Writer, renderingContext.Control);
+
+        renderingContext.Writer.RenderEndTag();
+      }
+      else if (arguments.ColumnsWithValidationFailures[renderingContext.ColumnIndex])
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, CssClasses.ValidationErrorMarker);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        var iconInfo = new IconInfo(_resourceUrlFactory.CreateThemedResourceUrl(typeof(InfrastructureResourceUrlFactory), ResourceType.Image, c_validationIndicatorPlaceholderIcon).GetUrl());
+        iconInfo.Render(renderingContext.Writer, renderingContext.Control);
+
+        renderingContext.Writer.RenderEndTag();
+      }
+
       RenderCellContents(renderingContext, arguments);
 
+      if (validationFailures.Count > 0)
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, _cssClasses.CssClassScreenReaderText);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+        renderingContext.Writer.WriteEncodedText(validationInfoText);
+        renderingContext.Writer.RenderEndTag();
+
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Ul);
```

Comment:
We need to verify this: it is possible the best way to handle this is by using aria-label instead of printing a separate span:
https://www.scottohara.me/blog/2020/05/02/labelled-lists.html

gut feeling: we should use aria-label. This result in the same or better UX for screenreader users.

alternative (more effort and no payoff in our scenario): annodate the span as aria-hidden, add an ID and link it to the UL via aria-labeledby

Reply:
The only reason for the span here is that we have this structure of UL already for the row/validation column and it allows us to reuse the error parsing logic without adding a special case.

I am unsure what exactly you are getting at with the aria-labels. We would have the same problem in the previously rendered lists for the row/validation column wouldn't we?

Reply:
As discussed: `aria-label`